core/txdb/snapshot.go

```diff
@@ -40,10 +40,12 @@ func DecodeSnapshot(data []byte) (*state.Snapshot, error) {
 		issuances[hash] = issuance.ExpiryMs
 	}
 
-	return &state.Snapshot{
+	result := &state.Snapshot{
 		Tree:      tree,
 		Issuances: issuances,
-	}, nil
+	}
+	copy(result.InitialBlockHash[:], storedSnapshot.InitialBlockHash)
```

Comment:
Do we need it on the serialized snapshot proto? For backwards compatibility, I think we're going to need to lookup the initial block hash because any existing snapshots won't have it populated.